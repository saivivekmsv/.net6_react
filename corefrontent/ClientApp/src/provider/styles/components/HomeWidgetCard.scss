.home-widget-card {
  position: relative;
  background-color: #ffffff;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 15px;
  transition: transform 0.2s;

  height: 287px;
  width: 300px;
  cursor: pointer;
  text-decoration: none;
  color: inherit;
  margin: 20px;
  border: 1px solid #f2f2f2;

  .img-box {
    height: 200px;
    width: 280px;
    border-radius: 15px;
    margin-top: 5px;
  }

  .Company {
    background: url("/assets/icons/gif/Company.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Company.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Plan {
    background: url("/assets/icons/gif/Plan.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Plan.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Employees {
    background: url("/assets/icons/gif/Employees.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Employees.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Eligibility {
    background: url("/assets/icons/gif/Eligibility.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Eligibility.gif") no-repeat center
          center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Enrollment {
    background: url("/assets/icons/gif/Enrollment.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Enrollment.gif") no-repeat center
          center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Payroll {
    background: url("/assets/icons/gif/Payroll.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Payroll.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Reports {
    background: url("/assets/icons/gif/Reports.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Reports.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Mapper {
    background: url("/assets/icons/gif/Templates.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Templates.gif") no-repeat center center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Calculators {
    background: url("/assets/icons/gif/Calculator.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Calculator.gif") no-repeat center
          center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .Maintenance {
    background: url("/assets/icons/gif/Maintenance.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Maintenance.gif") no-repeat center
          center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .OutputFiles {
    background: url("/assets/icons/gif/Outputfile.svg") no-repeat center center,
      linear-gradient(
        116.36deg,
        rgba(81, 203, 242, 0.19) 2.11%,
        rgba(81, 203, 242, 0.1) 2.12%,
        rgba(81, 203, 242, 0.1) 97.98%
      );

    &:hover {
      background: url("/assets/icons/gif/Outputfile.svg") no-repeat center
          center,
        linear-gradient(
          116.36deg,
          rgba(81, 203, 242, 0.19) 2.11%,
          rgba(81, 203, 242, 0.1) 2.12%,
          rgba(81, 203, 242, 0.1) 97.98%
        );
      background-size: 200px;
    }
  }

  .content-container {
    line-height: 35px;
    padding-left: 10px;
    font-size: 18px;
  }

  &.disabled {
    pointer-events: none;
    opacity: 0.5;
  }

  .widget-icon {
    padding: 10px;
    max-width: 100px;

    img {
      max-width: 100%;
    }
  }

  .widget-content {
    display: flex;
    flex-direction: column;
    align-self: flex-start;
    align-items: flex-start;

    .widget-label {
      font-size: 18px;
      line-height: 27px;
      padding-bottom: 10px;
      text-transform: capitalize;
    }

    .widget-count {
      font-weight: 300;
      font-size: 18px;
      line-height: 27px;
      text-transform: capitalize;
      font-feature-settings: "tnum" on, "lnum" on;
    }
  }

  .spinner-row {
    font-size: 7px;
  }

  .spinner-border-sm {
    height: 10px;
    width: 10px;
    border-width: 0.4em;
  }

  .add-icon {
    position: absolute;
    right: 10px;
    bottom: 10px;
  }

  @media only screen and (max-width: 1439px) {
    &:hover:not(.disabled) {
      transform: scale(1.1);
    }
  }
}
