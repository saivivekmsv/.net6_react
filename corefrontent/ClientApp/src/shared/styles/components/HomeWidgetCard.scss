.home-widget-card {
  display: flex;
  position: relative;
  background-color: #ffffff;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 4px;
  transition: transform 0.2s;
  align-items: center;
  height: 125px;
  max-width: 280px;
  cursor: pointer;
  text-decoration: none;
  color: inherit;
  border-left: 5px solid $primary;
  padding: 15px;
  &:hover:not(.disabled) {
    text-decoration: none;
    color: inherit;
    transform: scale(1.2);
  }
  &.disabled {
    pointer-events: none;
    opacity: 0.5;
  }
  .widget-icon {
    padding: 10px;
    max-width: 100px;
    img {
      max-width: 100%;
    }
  }
  .widget-content {
    display: flex;
    flex-direction: column;
    align-self: flex-start;
    align-items: flex-start;
    .widget-label {
      font-size: 18px;
      line-height: 27px;
      padding-bottom: 10px;
      text-transform: capitalize;
    }
    .widget-count {
      font-weight: 300;
      font-size: 18px;
      line-height: 27px;
      text-transform: capitalize;
      font-feature-settings: "tnum" on, "lnum" on;
    }
  }
  .spinner-row {
    font-size: 7px;
  }
  .spinner-border-sm {
    height: 10px;
    width: 10px;
    border-width: 0.4em;
  }
  .add-icon {
    position: absolute;
    right: 10px;
    bottom: 10px;
  }
  @media only screen and (max-width: 1439px) {
    &:hover:not(.disabled) {
      transform: scale(1.1);
    }
  }
}
