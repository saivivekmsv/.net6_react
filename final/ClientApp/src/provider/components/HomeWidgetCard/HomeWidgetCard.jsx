import React from "react";
import { Link } from "react-router-dom";
import { OverlayTrigger, Tooltip, Image } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlusSquare } from "@fortawesome/pro-light-svg-icons";

import { Loader } from "../";

const HomeWidgetCard = ({
  isDisabled,
  name,
  count,
  onAddClick,
  isLoading,
  link,
  showAdd,
  imgUrl,
}) => {
  const disabled = isDisabled ? "disabled" : "";
  return (
    <Link
      href="javascrip:void(0)"
      to={link}
      className={`home-widget-card d-flex flex-column justify-content-between align-items-center p-1 ${disabled}`}
    >
      <div className={`img-box ${name}`}></div>
      <div className="d-flex flex-column w-100 content-container">
        <div>
          <div className="widget-label">{name}</div>
          {isLoading && <Loader size="sm" />}
          {!isLoading && <div className="widget-count">{count}</div>}
        </div>
        <div>
          {showAdd && (
            <OverlayTrigger
              placement="bottom"
              overlay={<Tooltip>Add {name}</Tooltip>}
            >
              <button className="add-icon" onClick={onAddClick}>
                <FontAwesomeIcon icon={faPlusSquare} size="lg" />
              </button>
            </OverlayTrigger>
          )}
        </div>
      </div>
    </Link>
  );
};

export default HomeWidgetCard;
