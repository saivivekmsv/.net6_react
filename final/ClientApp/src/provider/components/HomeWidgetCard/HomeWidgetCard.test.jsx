import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import HomeWidgetCard from "./HomeWidgetCard";

describe("HomeWidgetCard", () => {
  const props = {
    name: "Plan",
    count: 270,
    onAddClick: () => {},
  };
  test("to match snapshot active", () => {
    const { baseElement } = render(
      <Router>
        <HomeWidgetCard {...props} />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("to match snapshot disabled", () => {
    const { baseElement } = render(
      <Router>
        <HomeWidgetCard {...props} isDisabled />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("to match snapshot loading", () => {
    const { baseElement } = render(
      <Router>
        <HomeWidgetCard {...props} isLoading />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("to match snapshot showAdd", () => {
    const { baseElement } = render(
      <Router>
        <HomeWidgetCard {...props} showAdd />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("to match snapshot no count", () => {
    const { baseElement } = render(
      <Router>
        <HomeWidgetCard {...props} showAdd count="" />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
