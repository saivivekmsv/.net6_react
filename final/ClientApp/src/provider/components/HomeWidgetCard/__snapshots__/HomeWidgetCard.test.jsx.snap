// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeWidgetCard to match snapshot active 1`] = `
<body>
  <div>
    <a
      class="home-widget-card "
      href=""
    >
      <div
        class="widget-icon"
      >
        <img
          alt="Plan"
          class=""
        />
      </div>
      <div
        class="widget-content"
      >
        <div
          class="widget-label"
        >
          Plan
        </div>
        <div
          class="widget-count"
        >
          270
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`HomeWidgetCard to match snapshot disabled 1`] = `
<body>
  <div>
    <a
      class="home-widget-card disabled"
      href=""
    >
      <div
        class="widget-icon"
      >
        <img
          alt="Plan"
          class=""
        />
      </div>
      <div
        class="widget-content"
      >
        <div
          class="widget-label"
        >
          Plan
        </div>
        <div
          class="widget-count"
        >
          270
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`HomeWidgetCard to match snapshot loading 1`] = `
<body>
  <div>
    <a
      class="home-widget-card "
      href=""
    >
      <div
        class="widget-icon"
      >
        <img
          alt="Plan"
          class=""
        />
      </div>
      <div
        class="widget-content"
      >
        <div
          class="widget-label"
        >
          Plan
        </div>
        <div
          class="spinner-row d-flex align-items-center"
        >
          <div
            class="spinner-border spinner-border-sm"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
            
          Loading
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`HomeWidgetCard to match snapshot no count 1`] = `
<body>
  <div>
    <a
      class="home-widget-card "
      href=""
    >
      <div
        class="widget-icon"
      >
        <img
          alt="Plan"
          class=""
        />
      </div>
      <div
        class="widget-content"
      >
        <div
          class="widget-label"
        >
          Plan
        </div>
        <div
          class="widget-count"
        />
      </div>
      <button
        class="add-icon"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-square fa-w-14 fa-lg "
          data-icon="plus-square"
          data-prefix="fal"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 64c8.8 0 16 7.2 16 16v352c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16V80c0-8.8 7.2-16 16-16h352m0-32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-60 206h-98v-98c0-6.6-5.4-12-12-12h-12c-6.6 0-12 5.4-12 12v98h-98c-6.6 0-12 5.4-12 12v12c0 6.6 5.4 12 12 12h98v98c0 6.6 5.4 12 12 12h12c6.6 0 12-5.4 12-12v-98h98c6.6 0 12-5.4 12-12v-12c0-6.6-5.4-12-12-12z"
            fill="currentColor"
          />
        </svg>
      </button>
    </a>
  </div>
</body>
`;

exports[`HomeWidgetCard to match snapshot showAdd 1`] = `
<body>
  <div>
    <a
      class="home-widget-card "
      href=""
    >
      <div
        class="widget-icon"
      >
        <img
          alt="Plan"
          class=""
        />
      </div>
      <div
        class="widget-content"
      >
        <div
          class="widget-label"
        >
          Plan
        </div>
        <div
          class="widget-count"
        >
          270
        </div>
      </div>
      <button
        class="add-icon"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-square fa-w-14 fa-lg "
          data-icon="plus-square"
          data-prefix="fal"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 64c8.8 0 16 7.2 16 16v352c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16V80c0-8.8 7.2-16 16-16h352m0-32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-60 206h-98v-98c0-6.6-5.4-12-12-12h-12c-6.6 0-12 5.4-12 12v98h-98c-6.6 0-12 5.4-12 12v12c0 6.6 5.4 12 12 12h98v98c0 6.6 5.4 12 12 12h12c6.6 0 12-5.4 12-12v-98h98c6.6 0 12-5.4 12-12v-12c0-6.6-5.4-12-12-12z"
            fill="currentColor"
          />
        </svg>
      </button>
    </a>
  </div>
</body>
`;
